doc: Cloud static IP docs (#11009)
MaterializeInc/materialize: doc/user/content/cloud/security.md

* Single sign-on integration (SSO)

In the future, the enterprise product will also provide secure network ingress and egress. If you have other specific security requirements, please [let us know](../support).

## Static IP addresses

All Materialize [Cloud deployments](../cloud-deployments/) come with a static IP address.

This gives you the ability to connect your Materialize Cloud deployments to [sources](../../sql/create-source) and [sinks](../../sql/create-sink/) secured with a firewall.

Allowing the static IP address enables the connection from your Materialize Cloud deployments to your sources and sinks.

The specific commands to allow the static IP address will vary depending on your operating system and firewall (e.g. `iptables`, `firewall-cmd`, UFW, AWS, Azure, and etc). Please refer to the appropriate documentation for your specific firewall.

### Getting the Static IP Address

> Once you’ve created a [deployment](../cloud-deployments/), to get the static IP address associated with your deployment you can follow these steps:

Author: Doesn't this get a bit confusing as it kind of indicates, that the user needs to do something in order to associate the IP with the deployment where but it is already associated?

> Once you’ve created a [deployment](../cloud-deployments/), to get the static IP address associated with your deployment you can follow these steps:

Reviewer: Yeah, exactly, the static IP is already associated with the deployment; you just need to determine it.

> Once you’ve created a [deployment](../cloud-deployments/), to get the static IP address associated with your deployment you can follow these steps:

Reviewer 2: Ah, you're right; I got the wrong idea skimming it! You could phrase it as:

Once you've created a deployment, you can get _the/its_ associated IP address following these steps.

The original is fine, too, just reads a bit strange.

> Once you’ve created a [deployment](../cloud-deployments/), to get the static IP address associated with your deployment you can follow these steps:

Reviewer: Your latest suggestion sounds much clearer!

> Once you’ve created a [deployment](../cloud-deployments/), to get the static IP address associated with your deployment you can follow these steps:

Author: Sounds good! Thanks, @morsapaes! Just pushed that change

1. Click on a deployment to see the individual deployment view.
2. On the right-hand side, click on the copy button to copy the IP address.

### Allowing the Static IP Address

> The following is a list of commands to allow the static IP address with some of the most common firewalls.

Reviewer 2: ```suggestion
The following is a list of commands to allow the static IP address with some of the most common firewalls:
```

> #### IPTABLES

Reviewer 2: Could these be cased as whatever the original casing is (_e.g._ iptables), instead of defaulting to upper case?

```
sudo iptables -I INPUT -s <static IP> -j ACCEPT
```

#### FIREWALL-CMD

```
sudo firewall-cmd --zone=public --add-source=<static IP>
```

#### UFW

```
sudo ufw allow from <static IP>
```

#### CSF

```
sudo csf -a <static IP>
```

#### AWS Security Groups

> If you are already using AWS Security Groups, you can add the static IP address to the security group to allow connections from your Materialize Cloud deployments.
> 
> To add the static IP address to your AWS Security Group, follow these steps:

Reviewer 2: ```suggestion
If you're already using AWS security groups, you can add the static IP address to an existing security group to allow connections from your Materialize Cloud deployments.

To add the static IP address to your AWS security group, follow these steps:
```

1. Log into your AWS account.
1. Go to your **Security Groups** and click on the **Security Group** you want to add the static IP address to.
1. Click on the **Edit Inbound Rules** button.
1. Click on the **Add rule** button.
1. Select **Custom TCP** as the type.
1. Specify the **Port range** depending on the source you want to connect to. For example, if you want to connect to a Kafka source, you would specify **Port range: 9092** or if you want to connect to a PostgreSQL source, you would specify **Port range: 5432**.

> 1. Add the Materialize Cloud deployment’s static IP address followed by `/32` to the **Source** field.

Reviewer 2: It would be nice to include the "why" for the `/32` in ().

Author: Just added a small explanation. Though I hope that most people who would be fiddling around with firewall rules should know what `/32` means

1. Click on the **Save** button.

> For more information on AWS Security Groups, please refer to the AWS [VPC Security Groups documentaiton](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html).

Reviewer 2: ```suggestion
For more information on AWS security groups, please refer to the AWS [VPC security groups documentation](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html).
```

> #### Azure

Reviewer 2: Nit, but it's weird (as in, inconsistent) to specify "AWS security groups", then just "Azure" and "GCP".

Author: Good point! Just updated this

> For more information on how to allow the static IP address with Azure, please refer to the [Azure documentation](https://docs.microsoft.com/en-us/azure/virtual-network/ip-services/configure-public-ip-firewall)

Reviewer 2: ```suggestion
For the steps required to allow static IP addresses with Azure, please refer to the [Azure documentation](https://docs.microsoft.com/en-us/azure/virtual-network/ip-services/configure-public-ip-firewall).
```

#### GCP

> For more information on how to allow the static IP address with GCP, please refer to the [GCP documentation](https://cloud.google.com/vpc/docs/using-firewalls)

Reviewer 2: ```suggestion
For the steps required to allow static IP addresses with GCP, please refer to the [GCP documentation](https://cloud.google.com/vpc/docs/using-firewalls).
```
